Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp

```diff
@@ -326,3 +326,30 @@ void webkit_web_view_toggle_inspector(WebKitWebView* webView)
         inspector->show();
 }
 #endif
+
+/**
+ * webkit_web_view_get_theme_color:
+ * @web_view: a #WebKitWebView
+ * @color: (out): a #WebKitColor to fill in with the theme color
+ *
+ * Gets the theme color that is specified by the content in the @web_view.
+ *
+ * Returns: Whether the currently loaded page defines a theme color.
+ *
+ * Since: 2.50
+ */
+gboolean
+webkit_web_view_get_theme_color(WebKitWebView* webView, WebKitColor* color)
+{
+    g_return_val_if_fail(WEBKIT_IS_WEB_VIEW(webView), false);
+    auto& page = webkitWebViewGetPage(webView);
+
+    if (!page.themeColor().isValid()) {
+        WebCore::Color tmpColor(WebCore::Color::transparentBlack);
+        webkitColorFillFromWebCoreColor(tmpColor, color);
```

Comment:
This should be documented. Why do we fill the color with transparent black? I would expect the color not to be filled at all, but if we want to do this we should say it in the documentation.

Reply:
> This should be documented. Why do we fill the color with transparent black? I would expect the color not to be filled at all, but if we want to do this we should say it in the documentation.

Because when you get the property using i.e. `g_object_get()` there is no way of telling the caller “there was't a color defined”, so one needs to fill the returned color with something. WebCore's `Color` class uses transparent black in that case, which IMO it makes sense in this case: if one “sees“ a transparent color, that would not really make much sense for a theme color, so that's kind of a neutral value for an invalid color.

Reply:
Makes sense, let's just document it then.